backend/src/email/middlewares/email-template.middleware.ts

```diff
@@ -379,10 +386,103 @@ const uploadProtectedCompleteHandler = async (
   }
 }
 
+/**
+ * Downloads the file from the vault url and checks that its columns match the attributes provided in the template.
+ * If a template has not yet been uploaded, do not write to the message logs, but prompt the user to upload a template first.
+ * If the template and csv do not match, prompt the user to upload a new file.
+ * @param req
+ * @param res
+ * @param next
+ */
+const tesseractHandler = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+): Promise<Response | void> => {
+  const { campaignId } = req.params
+  const { url } = req.body
+  const logMeta = { campaignId, action: 'tesseractHandler' }
+
+  try {
+    // check if template exists
+    const template = await EmailTemplateService.getFilledTemplate(+campaignId)
+    if (template === null) {
+      throw new Error(
+        'Error: No message template found. Please create a message template before uploading a recipient file.'
+      )
+    }
+
+    // Return early because bulk insert is slow
+    res.sendStatus(202)
+
+    try {
+      await retry(async () => {
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        const transaction = await Campaign.sequelize!.transaction()
+
+        // stream response from s3 presigned url
```

Comment:
I think `s3 presigned url` should be renamed to `Vault URL` or something along those lines?